### ISSUE TITLE
use get_url module instead of shelling out to "curl" 

### ISSUE THREAD

**Issue by vasukulkarni** [ceph/ceph-ansible#1299]
on xenial we see the following issue due to missing curl, the playbook needs fix there

2017-02-09T00:47:57.490 INFO:teuthology.orchestra.run.vpm031.stdout:TASK [ceph.ceph-common : fail if systemd is not present] ***********************
2017-02-09T00:47:57.491 INFO:teuthology.orchestra.run.vpm031.stdout:task path: /home/ubuntu/ceph-ansible/roles/ceph-common/tasks/checks/check_system.yml:38
2017-02-09T00:47:57.601 INFO:teuthology.orchestra.run.vpm031.stdout:fatal: [vpm031.front.sepia.ceph.com]: FAILED! => {"changed": false, "failed": true, "msg": "Systemd must be present"}
2017-02-09T00:47:57.616 INFO:teuthology.orchestra.run.vpm031.stdout:fatal: [vpm101.front.sepia.ceph.com]: FAILED! => {"changed": false, "failed": true, "msg": "Systemd must be present"}
2017-02-09T00:47:57.627 INFO:teuthology.orchestra.run.vpm031.stdout:fatal: [vpm151.front.sepia.ceph.com]: FAILED! => {"changed": false, "failed": true, "msg": "Systemd must be present"}

**Comment by vasukulkarni** [comment]
http://qa-proxy.ceph.com/teuthology/vasu-2017-02-09_00:21:42-ceph-ansible-jewel---basic-vps/799396/teuthology.log

**Comment by leseb** [comment]
On which distro are you running this? We do not support systems that are not running systemd.

**Comment by vasukulkarni** [comment]
Sorry this issue was discussed in #octo, The issue i believe is due to this https://github.com/ceph/ceph-ansible/commit/f741ee74  and it expects curl to be present on the system and eventually the repo get fails and the error is bit misleading. @ktdreyer  and @andrewschoen can provide more info on this issue

**Comment by ktdreyer** [comment]
The problem is that we use `curl` to download a repo config file from shaman, rather than using Ansible's built-in `get_url` module. This means `curl` has to be installed on all systems that download builds from shaman.

According to Alfredo's comment in `debian_ceph_repository.yml` and `redhat_ceph_repository.yml`:

```
# we must use curl instead of ansible's uri module because SNI support in
# Python is only available in 2.7.9 and later, and most supported distributions
# don't have that version, so a request to https fails.
```

... so te reason we shell out to `curl` rather than using a built-in Ansible module is because older distros' Pythons do not support TLS SNI. After some experiments, we found that Python on Ubuntu Trusty is the one that is still affected by this. RHEL 7 and Ubuntu Xenial can handle SNI just fine with the `get_url` module.

Before this week, https://shaman.ceph.com used TLS SNI to serve the x509 cert for `CN=shaman.ceph.com`. I've modified the nginx configuration there so that the web server issues the `CN=shaman.ceph.com` x509 cert first to non-SNI clients. Today non-SNI clients can successfully make requests.

This means that Ubuntu Trusty can use the `get_url` module with shaman now, and we can stop shelling out to `curl`.
